support optional TOTP for authentication

JanZerebecki opened the pull request:
pam_oath should be enabled in the cockpit pam config by appending:

auth [user_unknown=ignore success=ok] pam_oath.so usersfile=${HOME}/.pam_oath_usersfile window=20 digits=6

AdamMajer commented on tools/cockpit.spec:
```diff
@@ -238,6 +238,8 @@ BuildRequires:  python3-tox-current-env
 
 cp %SOURCE1 tools/cockpit.pam
+cp %SOURCE1 tools/cockpit.pam-2fa
+echo 'auth required pam_oath.so usersfile=${HOME}/.pam_oath_usersfile window=20 digits=6' >> tools/cockpit.pam-2fa
```
Can we just edit the `cockpit.pam` instead? We have it in the package. This should be set to optional (`nullok`) and then the admin can enforce it with manual override.

JanZerebecki replied:
@Vogtinator asked for a solution that a) does not require jeos-firstboot to mess with the content of pam config b) allows the cockpit package to upgrade pam config configuration. Having the admin to adjustments to the content means the package can not easily upgrade the content anymore.

JanZerebecki committed:
support optional TOTP for authentication

pam_oath should be enabled in the cockpit pam config by appending:

auth [user_unknown=ignore success=ok] pam_oath.so usersfile=${HOME}/.pam_oath_usersfile window=20 digits=6
